File: backend/src/main/java/ecrow/backend/business/abstracts/AddressService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.dtos.AddressAddDto;
import ecrow.backend.entities.dtos.AddressUpdateDto;

import java.util.List;

public interface AddressService {
    DataResult<List<Address>> getAll();
    DataResult<Address> getById(Integer id);
    DataResult<List<Address>> getByFkCustomerId(Integer customerId);
    Result deleteById(Integer id);


    Result add(AddressAddDto addressAddDto);
    Result update(AddressUpdateDto addressUpdateDto);

}

File: frontend/src/services/AddressService.js
import axios from "axios"

export default class AddressService{

    getAll(){
       return axios.get("http://localhost:8080/api/address/getAll")
    }

    getById(id){
        return axios.get(`http://localhost:8080/api/address/getById?id=${id}`)
    }

    getByCustomerId(customerId){
        return axios.get(`http://localhost:8080/api/address/getByFkCustomerId?customerId=${customerId}`)
    }

    deleteById(id){
        return axios.delete(`http://localhost:8080/api/address/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/address/add", values)
    }

    update(values){
        return axios.patch("http://localhost:8080/api/address/update", values)
    }
}